rust/rope/src/rope.rs

```diff
@@ -634,18 +634,27 @@ impl Rope {
         leaf.as_bytes()[pos]
     }
 
-    // TODO: this should be a Cow
-    // TODO: a case can be made to hang this on Cursor instead
-    pub fn slice_to_string<T>(&self, range: T) -> String 
+    pub fn slice_to_string<T>(&self, range: T) -> Cow<str>
```

Comment:
I think we should also rename this function, since it no longer returns a `String`. I could imagine calling it `slice_to_cow` or `get_slice` or `slice_cow` or `cow_slice`; even maybe `slice_str`?